frontend/app/src/components/account-management/create-account/premium/CreateAccountPremiumForm.vue

```diff
@@ -0,0 +1,136 @@
+<script setup lang="ts">
+import useVuelidate from '@vuelidate/core';
+import { helpers, requiredIf } from '@vuelidate/validators';
+import { toMessages } from '@/utils/validation';
+import { type PremiumSetup } from '@/types/login';
+
+const props = defineProps<{
+  loading: boolean;
+  enabled: boolean;
+  form: PremiumSetup;
+}>();
+
+const emit = defineEmits<{
+  (e: 'update:form', form: PremiumSetup): void;
+  (e: 'update:valid', valid: boolean): void;
+}>();
+
+const { t } = useI18n();
+
+const { enabled, form } = toRefs(props);
+
+const apiKey = computed({
+  get() {
+    return get(form).apiKey;
+  },
+  set(value: string | null) {
+    input({ apiKey: value?.trim() ?? '' });
+  }
+});
+
+const apiSecret = computed({
+  get() {
+    return get(form).apiSecret;
+  },
+  set(value: string | null) {
+    input({ apiSecret: value?.trim() ?? '' });
+  }
+});
+
+const syncDatabase = computed({
+  get() {
+    return get(form).syncDatabase;
+  },
+  set(value: boolean) {
+    input({ syncDatabase: value });
+  }
+});
+
+watch(enabled, enabled => {
+  if (!enabled) {
+    set(syncDatabase, false);
+  }
+});
+
+const onApiKeyPaste = (_event: ClipboardEvent) => {
+  const paste = trimOnPaste(_event);
+  if (paste) {
+    set(apiKey, paste);
+  }
+};
+
+const onApiSecretPaste = (_event: ClipboardEvent) => {
+  const paste = trimOnPaste(_event);
+  if (paste) {
+    set(apiSecret, paste);
+  }
+};
+
```

Comment:
I don't think we actually need this, couldn't we easily do the same thing by just doing v-model.trim?